added find_member methods

krissi opened the pull request:
This requests adds a `#find_member` method

krissi committed:
added find_member methods

cap10morgan commented:
Thanks for the PR! As a small style consistency thing, can you remove the `then` from the multi-line if statements?

cap10morgan commented on lib/mailmanager/lib.rb:
```diff
@@ -31,6 +31,13 @@ def list_names
     end
 
+    def find_member(regex)
+      regex = regex.source if regex.is_a? Regexp
```
Does Mailman support all the same regex semantics as Ruby?

krissi committed:
Raise when a Regexp is given to #find_member, add specs

krissi commented:
Removed the `then`s

Removed the automagic conversion of the Regexp and added an exception if one is passed to let the user decide if the regex is valid.

I also added some specs
